background.js

```diff
@@ -1,29 +1,62 @@
+// Function to get the current tab
+async function getCurrentTab() {
+    let queryOptions = { active: true, currentWindow: true };
+    let [tab] = await chrome.tabs.query(queryOptions);
+    return tab;
+}
 
-// Called when the user clicks on the browser action.
-chrome.browserAction.onClicked.addListener(function(tab) {
-    var link = '[' + tab.title + '](' + tab.url + ')';
+function injectedFunction(tab) {
```

Comment:
- We can use `async function` to avoid changing `git blame` result
- Although set badge feature seems to broken, I would like to leave it untouched. Maybe I can fix that later.

Reply:
```javascript
async function injectedFunction(tab) {
    title = tab.title;
    
    const storage = await chrome.storage.sync.get(null)
    for (const index in storage.fieldsets) {
        regexUrl = new RegExp(storage.fieldsets[index].url, 'gi');
        if (regexUrl.test(tab.url)) {
            regexSearch = new RegExp(storage.fieldsets[index].search, 'gi');
            regexReplace = new RegExp(storage.fieldsets[index].replace, 'i');
            if (regexSearch.test(title)) {
                title = title.replace(regexSearch, storage.fieldsets[index].replace);
                // we don't stop in case there are several rules
            }
        }
    }

    var link = '[' + tab.title + '](' + tab.url + ')';

    element = document.createElement("textarea");
    element.value = link;

    // It's required to put the temporary element to document tree before copy command
    // See: https://stackoverflow.com/questions/25622359/clipboard-copy-paste-on-content-script-chrome-extension
    document.body.appendChild(element);
    element.select();
    document.execCommand('copy');

    element.remove();

    chrome.browserAction.setBadgeText({
        text: "Done",
        tabId: tab.id
    });

    var clear_callback = function () {
        chrome.browserAction.setBadgeText({
            text: "",
            tabId: tab.id
        });
    };
    window.setTimeout(clear_callback, 1500);
}
```